### PR TITLE
feat/retain message id if exists

### PR DESCRIPTION
**Pull request by drake-888**
- [ ] I've included a changeset (psst. run `yarn changeset`. Read about changesets [here](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md)).

### CODE REVIEW

**Comment by RoshniChan** [comment]
Hello @silesky , Could you spare a moment to look into this?

File: packages/core/src/events/index.ts [hunk 1]
```diff
@@ -271,7 +271,7 @@ export class EventFactory {
 
     const evt: CoreSegmentEvent = {
       ...body,
-      messageId: this.createMessageId(),
+      messageId: (event.messageId) ? event.messageId : this.createMessageId(),
```
Comment (silesky) [inline packages/core/src/events/index.ts:274]: event.messageId || this.createMessageId() would be more idiomatic IMO.

File: .changeset/message-id-backward-compatibility.md
Comment (silesky) [inline .changeset/message-id-backward-compatibility.md:1]: Thanks for your contribution! Could you write tests!
?

File: packages/core/src/events/index.ts (hunk 1 above)
Comment (drake-888) [reply packages/core/src/events/index.ts]: @silesky , updated code base & added it into validateEvent.

File: .changeset/message-id-backward-compatibility.md
Comment (drake-888) [reply .changeset/message-id-backward-compatibility.md:1]: added. let me know if there is further action needed.

**Comment by silesky** [comment]
Hey @RoshniChan @drake-888, I've moved the work into: https://github.com/segmentio/analytics-next/pull/1043

Thanks for getting it started @drake-888.

**Pull request closed without merging**